Add note about required-ros-distributions caveat on Ubuntu

christophebedard opened the pull request:
As suggested here: https://github.com/ros-tooling/setup-ros/issues/439#issuecomment-1025247201

Signed-off-by: Christophe Bedard <bedard.christophe@gmail.com>

christophebedard commented:
@amacneil let me know if this looks OK to you!

emersonknapp commented:
I feel like this note could be a little more concise. Note that https://github.com/ros-tooling/setup-ros/blob/master/action.yml#L19 documents this feature (`action.yml` is essentially the "API docs" for an action). Maybe this should just be a bold note that **"required-ros-distributions installs the desktop variant for that distribution. This option is not required, and should probably be avoided in real workflows. It is retained for historical reasons and those who specifically do not care about whether their application specifies its dependencies properly."**

christophebedard commented:
Sure, I changed it to that: 3b239dc17ff48bd1e30aa45c43b4c4a411679499

I also updated the `action.yml` file to mention that `required-ros-distributions` installs binaries on Windows: 3234b26bdcb3f352e6fc8c2aa44f3e202459ed82
